src/Simulation/Simulators/ResourcesEstimator/ResourcesEstimatorWithAdditionalPrimitiveOperations.cs

```diff
@@ -0,0 +1,92 @@
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.QCTraceSimulatorRuntime;
+using Microsoft.Quantum.Simulation.Simulators;
+using Microsoft.Quantum.Simulation.Simulators.QCTraceSimulators;
+
+namespace Simulator
+{
+    public class ResourcesEstimatorWithAdditionalPrimitiveOperations : ResourcesEstimator
+    {
+        public ResourcesEstimatorWithAdditionalPrimitiveOperations() : this(ResourcesEstimator.RecommendedConfig())
+        {
+        }
+
+        public ResourcesEstimatorWithAdditionalPrimitiveOperations(QCTraceSimulatorConfiguration config) : base(WithoutPrimitiveOperationsCounter(config))
+        {
+        }
+
+        private static QCTraceSimulatorConfiguration WithoutPrimitiveOperationsCounter(QCTraceSimulatorConfiguration config)
+        {
+            config.UsePrimitiveOperationsCounter = false;
+            return config;
+        }
+
+        protected virtual IDictionary<string, IEnumerable<Type>>? AdditionalOperations { get; }
```

Comment:
This was intentionally made a property to be generic for arbitrary additional operations. I think it's okay to hard code the additional operators and not providing this property when merging it with the `ResourcesEstimator` class.